packages/@lightningjs/ui-components/src/mixins/withThemeStyles/utils.js

```diff
@@ -0,0 +1,534 @@
+
+      if (component._componentLevelStyle?.mode) {
+        finalStyle = clone(finalStyle, {
+          mode: component._componentLevelStyle.mode
+        });
+        const [styleModeTone, styleModeToneDelete] = styleFormatter(
+          finalStyle,
+          'mode',
+          'tone'
+        );
+        if (styleModeTone) {
+          finalStyle = clone(finalStyle, { tone: styleModeTone });
+          deleteNestedProperty(finalStyle, styleModeToneDelete);
+        }
+      }
+
+      if (component._componentLevelStyle) {
+        const cleanStyle = JSON.parse(
+          JSON.stringify(component._componentLevelStyle)
+        );
+        delete cleanStyle.tone;
+        delete cleanStyle.mode;
+        finalStyle = clone(finalStyle, { base: cleanStyle });
+        for (const key in finalStyle?.tone) {
+          finalStyle = clone(finalStyle, { tone: { [key]: cleanStyle } });
+        }
+        for (const key in finalStyle?.mode) {
+          finalStyle = clone(finalStyle, { mode: { [key]: cleanStyle } });
+        }
+      }
+    }
+  });
+  // Process style object
+  const processedStyle = JSON.stringify(finalStyle, (_, value) => {
+    if ('string' === typeof value && value.startsWith('theme.')) {
+      // Support theme strings example: theme.radius.md
+      return getValFromObjPath(component, value); // If no theme value exists, the property will be removed from the object
+    } else if (Array.isArray(value) && value.length === 2) {
+      // Process value as a color ['#663399', 1]
+      return getHexColor(value[0], value[1]);
+    }
+    return value;
+  });
+
+  return removeEmptyObjects(JSON.parse(processedStyle));
+};
+
+/**
+ * Generates the final style object for a component using its style source.
+ * @param {object} component - The component for which to generate a style object.
+ * @param {object} componentStyleSource - The style source object for the component.
+ * @returns {object} - The generated style object.
+ */
+export const generateStyle = (component, componentStyleSource = {}) => {
+  if (!isPlainObject(component)) return {};
+  const { mode = 'unfocused', tone = 'neutral' } = component;
+
+  const modeStyle = componentStyleSource?.mode?.[mode];
+  const toneStyle = componentStyleSource?.tone?.[tone];
+
+  // Start with base styles
+  let finalStyle = componentStyleSource.base || {};
+
+  // Add tone style
+  if (toneStyle) {
+    finalStyle = clone(finalStyle, toneStyle);
+  }
+
+  // Add mode style
+  if (modeStyle) {
+    finalStyle = clone(finalStyle, modeStyle);
+  }
+
+  return finalStyle;
+};
+
+/**
+ * Traverse up the prototype chain to create an array of all the styles that are present in the Components ancestors
+ * @param {object} componentObj - The component object to get the style chain from.
+ * @returns {{ style: (object | function) }[]} - An array of style objects containing either an object of styles or a function to return an object of styles.
+ */
+export function getStyleChain(componentObj) {
```

Comment:
if you're going to set all the other functions as variables, might as well do it here too for consistency

Reply:
Changed to match